hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -2150,12 +2150,8 @@ private void acceptPlan(final HashMap<RegionInfo, RegionStateNode> regions,
     final ProcedureEvent<?>[] events = new ProcedureEvent[regions.size()];
     final long st = System.currentTimeMillis();
 
-    if (plan == null) {
-      throw new HBaseIOException("unable to compute plans for regions=" + regions.size());
-    }
-
     if (plan.isEmpty()) {
-      return;
+      throw new HBaseIOException("unable to compute plans for regions=" + regions.size());
```

Comment:
Why this change?

Reply:
we use empty Map replace null return, so it means exception occur during assignment

Reply:
Oh, just saw the 'plan == null' in the old code. This is a bit strange as the plan is a function parameter, but anyway, not your fault.